librdmacm/rsocket.c

```diff
@@ -1345,6 +1349,10 @@ int raccept(int socket, struct sockaddr *addr, socklen_t *addrlen)
 
 	if (addr && addrlen)
 		rgetpeername(new_rs->index, addr, addrlen);
+	/* The app can still drive the CM state on failure */
+	int save_errno = errno;
+	rs_notify_svc(&accept_svc, new_rs, RS_SVC_ADD_CM);
+	errno = save_errno;
```

Comment:
The commit message needs more detail of the problem and how the change helps.  For example, see:

https://github.com/linux-rdma/rdma-core/commit/659c4f903a2aec8c60adfbb5a315b107e84acdd5

for when the connect_svc was added. I suspect the existing connect_svc can be re-used here, which avoids the possibility of starting up another thread, and is already coded for cleanup, which is missing from this patch.

Reply:
The example is very useful for me to unstand connect svc!! thank very much. I think it can be reused to handle cm event for acceptor new_rs. And while reusing connect_svc, i think  no more cleanup codes is needed, right?

And the commit message is modified, please take a look

Reply:
Correct - cleanup is already there for the connect_svc.  Commit message is better.  It may be worth referencing the original commit for the cleanup svc as part of the message, but someone interested could use git blame to find it fairly quickly.

Reply:
Agreed, added refer for the original commit 659c4f9 in commit message.

Wondering whether the service name "connect_svc" would make someone else confused or not? As it's not only used by connect api calls, but also raccept reusing it now. Is there any suggestion or just keep this at present?

Reply:
It's managing the connection state, so I wouldn't bother changing the name.

Reply:
got it